cdap-examples/DataCleansing/src/test/java/co/cask/cdap/examples/datacleansing/DataCleansingMapReduceTest.java

```diff
@@ -168,21 +185,23 @@ private void compareData(Long time, String dsName, Set<String> expectedRecords)
 
   private Set<String> getDataFromFile(Long time, String dsName) throws Exception {
     DataSetManager<PartitionedFileSet> cleanRecords = getDataset(dsName);
-    PartitionDetail partition =
-      cleanRecords.get().getPartition(PartitionKey.builder().addLongField("time", time).build());
-
-    if (partition == null) {
-      return Collections.emptySet();
-    }
+    PartitionFilter filter = PartitionFilter.builder().addValueCondition("time", time).build();
+    return getDataFromFilter(cleanRecords.get(), filter);
+  }
 
+  private Set<String> getDataFromFilter(PartitionedFileSet partitionedFileSet,
+                                        PartitionFilter filter) throws IOException {
+    Set<PartitionDetail> partitions = partitionedFileSet.getPartitions(filter);
     Set<String> cleanData = new HashSet<>();
-    Location partitionLocation = partition.getLocation();
-    for (Location location : partitionLocation.list()) {
-      if (location.getName().startsWith("part-")) {
-        try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(location.getInputStream()))) {
-          String line;
-          while ((line = bufferedReader.readLine()) != null) {
-            cleanData.add(line);
+    for (PartitionDetail partition : partitions) {
+      Location partitionLocation = partition.getLocation();
+      for (Location location : partitionLocation.list()) {
+        if (location.getName().startsWith("part-")) {
```

Comment:
This is existing code, but is it safe to make the assumption that all files start with ``part-``? This is controlled by the configuration [mapreduce.output.basename](https://github.com/apache/hadoop/blob/trunk/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/output/FileOutputFormat.java#L50). Its default value is ``part`` and you get its current value by using [FileOutputFormat.getOutputName()](https://github.com/apache/hadoop/blob/trunk/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/output/FileOutputFormat.java#L293).